packages/fxa-auth-server/lib/routes/subscriptions.js

```diff
@@ -0,0 +1,272 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+const error = require('../error');
+const isA = require('joi');
+const ScopeSet = require('fxa-shared').oauth.scopes;
+
+module.exports = (log, db, config, customs, oauthdb, subhub) => {
+  const totpUtils = require('./utils/totp')(log, config, db);
+
+  const SUBSCRIPTIONS_MANAGEMENT_SCOPE =
+    'https://identity.mozilla.com/apps/subscriptions';
+
+  async function handleAuth(auth) {
+    let scope, uid, email;
+
+    if (auth.strategy === 'sessionToken') {
+      // sessionToken strategy
+      scope = { contains: () => true };
+
+      const sessionToken = auth.credentials;
+      ({ uid, email } = sessionToken);
+
+      const hasEnabledToken = await totpUtils.hasTotpToken({ uid });
+      if (
+        hasEnabledToken
+        && (
+          sessionToken.tokenVerificationId
+          || sessionToken.authenticatorAssuranceLevel <= 1
+        )
+      ) {
+        throw error.unverifiedSession();
+      }
+    } else {
+      // oauthToken strategy
+      scope = ScopeSet.fromArray(auth.credentials.scope);
+      if (! scope.contains(SUBSCRIPTIONS_MANAGEMENT_SCOPE)) {
+        throw error.invalidToken('Invalid authentication scope in token');
+      }
+      uid = auth.credentials.user;
+
+      const account = await db.account(uid);
+      email = account.primaryEmail.email;
+    }
+
+    return { scope, uid, email };
+  }
+
+  return [
+    {
+      method: 'POST',
+      path: '/subscriptions/token',
+      options: {
+        auth: {
+          strategy: 'sessionToken'
+        },
+        response: {
+          schema: oauthdb.api.grantTokensFromCredentials.opts.validate.response
+        },
+      },
+      handler: async function (request) {
+        log.begin('subscriptions.grantToken', request);
+        return oauthdb.grantTokensFromSessionToken(
+          request.auth.credentials,
+          {
+            grant_type: 'fxa-credentials',
+            access_type: 'online',
+            scope: `profile:email ${SUBSCRIPTIONS_MANAGEMENT_SCOPE}`,
+            client_id: config.subscriptions.managementClientId,
+            ttl: config.subscriptions.managementTokenTTL
+          }
+        );
+      }
+    },
+    {
+      method: 'GET',
+      path: '/subscriptions/plans',
+      options: {
+        auth: {
+          strategies: [
+            'sessionToken',
+            'oauthToken'
+          ]
+        },
+        response: {
+          schema: isA.object().keys({
+            plan_id: isA.string().required(),
+            product_id: isA.string().required(),
+            interval: isA.string().required(),
+            amount: isA.string().required(),
+            currency: isA.string().required()
+          })
+        }
+      },
+      handler: async function (request) {
+        log.begin('subscriptions.listPlans', request);
+        await handleAuth(request.auth);
```

Comment:
Maybe? Should be fairly static results for display on the management page though. If we do rate limit, is that a thing we do elsewhere or maybe in the customs server? (looking for an example)

Reply:
FWIW, I think it is ok to not rate limiting here since we are using an authentication strategy. If the route was unauthenticated, we should definitely rate limit though.

Reply:
Filed a follow-up for this - https://github.com/mozilla/fxa/issues/884